### ISSUE TITLE
Improve calculation of resident cities feature

### ISSUE THREAD

**Issue by papjul** [breezy-weather/breezy-weather#58]
### Steps to reproduce

TBD

### Expected behavior

TBD

### Actual behavior

TBD

### Weather provider used

Open-Meteo, OpenWeather, etc, possibly all impacted

### Breezy Weather version

4.0.0-alpha

### Android version

All

### Device

All

### Other details

As reported by @LuccoJ, even AccuWeather may be impacted.

Currently, resident city feature probably work by comparing cityId. However, only AccuWeather and China providers rely on city ids. We need to implement it differently so that it relies on 5 km around instead.

Formula:
```java
/**
 * Calculate distance between two points in latitude and longitude taking
 * into account height difference. If you are not interested in height
 * difference pass 0.0. Uses Haversine method as its base.
 * 
 * lat1, lon1 Start point lat2, lon2 End point el1 Start altitude in meters
 * el2 End altitude in meters
 * @returns Distance in Meters
 */
public static double distance(double lat1, double lat2, double lon1,
        double lon2, double el1, double el2) {

    final int R = 6371; // Radius of the earth

    double latDistance = Math.toRadians(lat2 - lat1);
    double lonDistance = Math.toRadians(lon2 - lon1);
    double a = Math.sin(latDistance / 2) * Math.sin(latDistance / 2)
            + Math.cos(Math.toRadians(lat1)) * Math.cos(Math.toRadians(lat2))
            * Math.sin(lonDistance / 2) * Math.sin(lonDistance / 2);
    double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
    double distance = R * c * 1000; // convert to meters

    double height = el1 - el2;

    distance = Math.pow(distance, 2) + Math.pow(height, 2);

    return Math.sqrt(distance);
}
```

More investigation are to be made.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to **[latest version](https://github.com/breezy-weather/breezy-weather/releases/latest)**.

**Comment by papjul** [comment]
No longer a bug, as found out, it actually works by taking `0.8` degrees, however that’s way too much, and should use the formula above instead.

**Comment by papjul** [comment]
Implemented in eaebfa907828b0a6dd784f0213270885bf886ee3

Leaving open to make this configurable
